### PR TITLE
Update to pagination for ds6 to work with ebayui-core

### PR DESCRIPTION
**Pull request by martin-ebay**
small pagination fix to make fluid and non fluid work in ds6 with ebayui-core

## Description
a max of 8 items where showing up with the default usage.  and responsive sizes where not hiding the correct items.

## Context
fix to go along with changes to ebayui-core for 1.0.0 release.

### CODE REVIEW

**Comment by yomed** [comment]
I don't understand exactly. The DS6 fluid example doesn't seem to be fluid:
screen shot 2018-08-29 at 12 30 31 pm

**Comment by msendlakowski** [comment]
It looks like the demo for pagination--fluid may need to have the width set to 100%.  Once I do that, I can see that it flexes.

**Comment by seangates** [comment]
It's been that way since it was built. I thought the idea was to keep the spacing, and then it would wrap for smaller screens.

**Comment by martin-ebay** [comment]
I asked about the wrapping the other day in slack and was told it should only show the correct number of items as the component changes size, vs wrapping.

if we want wrapping if not fluid then these updates will need modified, and skin will need updated again.  In skin ds6 the pagination__items have overflow: hidden so anything that wraps gets hidden, which is what I was seeing before these changes.

**Comment by seangates** [comment]
@martin-ebay Let me rephrase.

I _think_ the idea was to keep the spacing, and then it would wrap for smaller screens.

**Comment by martin-ebay** [comment]
if you check out http://ebay.github.io/skin/ds6/#pagination and look at the second pagination sample and shrink your window you will see items wrap but are not visible.

So I was working off the info that this shouldn't be happening, and I should properly hide elements that would otherwise wrap.

File: src/less/pagination/ds6/pagination.less [hunk 1]
```diff
@@ -15,23 +16,17 @@
...
-}
-
-.pagination__items > li:not([hidden]) {
-    margin-right: 8px;
-}
-
-.pagination__items > li:last-child {
-    margin-right: 0;
+    > li:not([hidden]) {
+        width: 56px;
```
Comment (seangates) [inline src/less/pagination/ds6/pagination.less]: Could we just make them all 56px, whether they're visible or not? Does that affect anything?

File: src/less/pagination/ds6/pagination.less [hunk 2]
```diff
@@ -100,13 +95,7 @@
 .pagination--fluid {
     .pagination__items {
         flex-grow: 1;
-        height: 3rem; // a11y: needs to be relative to the text zoom level
         max-width: 512px;
-        padding: 4px 0 4px 8px;
-
-        > li {
-            line-height: 3rem; // a11y: needs to be relative to the text zoom level
```
Comment (seangates) [inline src/less/pagination/ds6/pagination.less]: We should ensure zoom still works.

**Comment by martin-ebay** [comment]
I was just overriding a rule from base I believe.  I tried without doing the !hidden bit but it wasn’t specific enough to override the rule.

File: src/less/pagination/ds6/pagination.less (hunk 2 above)
Comment (msendlakowski) [reply src/less/pagination/ds6/pagination.less]: Testing now

Comment (msendlakowski) [reply src/less/pagination/ds6/pagination.less]: Yes. Zoom is still working with this change.

**Comment by martin-ebay** [comment]
I was told that is what we wanted, and that being hidden after wrapping was not intentional

**Comment by seangates** [comment]
Oh ... nope.

**Comment by seangates** [comment]
What I believe I said was, we can wrap and hide in Skin, then remove these properties when the JS is invoked, and you're calculating which pages to hide based on available container width.

File: src/less/pagination/ds6/pagination.less (hunk 1 above)
Comment (martin-ebay) [reply src/less/pagination/ds6/pagination.less]: this is to override a style from the base/pagination.less.  I wasn't sure if that was being utilized differently for ds4.  If this isn't needed for ds4 I propose modifying the base/pagination and make it just 
```
> li { and remove the :not([hidden]);
```

**Comment by martin-ebay** [comment]
gotcha, I can work on that update.  should be just adding the overflow: hidden back in and making sure there is enough space for the button focus outlines

**Comment by msendlakowski** [comment]
yep.  DS6 needs 3rem and DS4 needs 2rem I think

**Comment by msendlakowski** [comment]
I think the fixed heights are going to be an a11y issue for text zooming.  Let me try it out.

**Comment by msendlakowski** [comment]
Can you add overflow hidden to the base class.  Its still wrapping in DS4.
screen shot 2018-08-30 at 1 57 46 pm

**Comment by msendlakowski** [comment]
Yeah, fixed height causes problems in firefox when the text is zoomed.  This appears to work, after moving overflow:hidden to the base css.

DS6:
```
.pagination__items {
    height: 3rem;
    padding: 4px 0;

    > li:not([hidden]) {
        width: 56px;
        line-height: 3rem;
    }
}
```

DS4:
```
.pagination__items {
    height: 2rem;
}
```

**Comment by seangates** [comment]
We're going to merge this to a branch we control, then we can make a couple tweaks to get it into 6.0.0.

Thanks, @martin-ebay!
